File: Android_Zone_Lib/src/com/zone/lib/TimeDiffUtils.java
package com.zone.lib;

import java.util.HashMap;
import java.util.Map;

/**
 * Created by fuzhipeng on 2016/10/24.
 */

public class TimeDiffUtils {


    static Map<String, Long> timeMaps = new HashMap<>();

    public static void start(String tag) {
        timeMaps.put(tag, System.currentTimeMillis());
    }

    public static long end(String tag) {
        Long startTime = timeMaps.get(tag);
        if (startTime == 0)
        throw  new IllegalStateException("please first use Method:start()");
            long diff = System.currentTimeMillis() -startTime;
        System.out.println("TimeDiffUtils--->"+tag+":"+diff);
        return diff;
    }

}

File: Android_Zone_Lib/src/com/zone/lib/TimeDiffUtils.kt
package com.zone.lib

import java.util.HashMap

/**
 * Created by zone on 2016/10/24.
 */

object TimeDiffUtils {
    internal var timeMaps: MutableMap<String, Long> = HashMap()

    fun start(tag: String) {
        timeMaps[tag] = System.currentTimeMillis()
    }

    fun end(tag: String): Long {
        val startTime = timeMaps[tag] ?: 0L
        if (startTime == 0L) throw IllegalStateException("please first use Method:start()")
        val diff = System.currentTimeMillis() - startTime
        LogZSDK.i("TimeDiffUtils--->$tag:$diff")
        return diff
    }
}
